pkg/cmd/repo/rename/http.go

```diff
@@ -0,0 +1,40 @@
+package rename
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+)
+
+func runRename(client *http.Client, repo ghrepo.Interface, newRepoName string) error {
+	input := map[string]string{"name": newRepoName}
+	body, err := json.Marshal(input)
+	if err != nil {
+		return err
+	}
+
+	path := fmt.Sprintf("%srepos/%s",
+		ghinstance.RESTPrefix(repo.RepoHost()),
+		ghrepo.FullName(repo))
+
+	request, err := http.NewRequest("PATCH", path, bytes.NewBuffer(body))
```

Comment:
This PATCH request is missing a `Content-Type` header. Since it's posting JSON data, the content type should be `application/json`. The server assumes JSON anyway, so it will accept this payload in any case, but it's best to be explicit about this in general.